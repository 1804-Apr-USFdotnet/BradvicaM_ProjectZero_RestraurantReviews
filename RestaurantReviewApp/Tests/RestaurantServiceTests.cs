using ApprovalTests;
using ApprovalTests.Reporters;
using Autofac;
using Console;
using Library.BusinessInterfaces;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Tests
{
    [TestClass]
    public class RestaurantServiceTests
    {
        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TopThreeRestaurantByAverageReview_Returns_CorrectRestaurants()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.TopThreeRestaurantByAverageReview();

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void AllRestaurants_Returns_AllRestaurantsFromDatabase()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.AllRestaurants();

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void SearchByString_Returns_AllRestaurantsThatContainStringValue()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.SearchByString("Dirty");

                Approvals.VerifyAll(results, "Restaurants");
            }
        }
    }
}
