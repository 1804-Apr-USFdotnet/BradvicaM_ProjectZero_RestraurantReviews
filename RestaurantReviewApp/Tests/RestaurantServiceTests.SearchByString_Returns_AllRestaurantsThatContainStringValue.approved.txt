Restaurants[0] = 
Id: 07ea7c51-070b-4ee9-9883-12443a96752b
Name: Dirty Devin's Mud Hut 
Street: 123 Dirty St.
City: Tampa
State: FL
ZipCode: 12341
PhoneNumber: 1234569131
Website: www.dirtydevins.com
Rating: 0

